x/ibc/relayer/README.md

```diff
@@ -0,0 +1,98 @@
+# Relayer
+
+The `relayer` package contains some basic relayer implemenations that are meant to be used by users wishing to relay packets between IBC enalbed chains. It is also well documented and intended as a place where users who are interested in building their own relayer can come for working examples.
+
+### Relayer Home Folder Layout 
+
+```bash
+~/.relayer
+├── chains
+│   ├── chain-id-1
+│   │   ├── keys
+│   │   │   └── keys.db
+│   │   └── verifiers
+│   │       └── chain-id-2.db
+│   └── chain-id-2
+│       ├── keys
+│       │   └── keys.db
+│       └── verifiers
+│           └── chain-id-1.db
+└── config
+    └── config.yaml
+```
+
+### Configuring the Relayer
+
+There are two major parts of `relayer` configuration:
+
+```go
+type Config struct {
+    Global Global `yaml:"global"`
+    Chains []Chain `yaml:"chains"`
+}
+```
+
+#### Global Configuration
+
+- Amount of time to sleep between relayer loops
+- Which strategy to use for your relayer (`naieve` is the only one implemented)
+- Home directory for the relayer (`~/.relayer`), to be passed in via flag
+
+```go
+type Global struct {
+    RelayTimeout string `yaml:"relay-timeout"`
+    Strategy     string `yaml:"strategy"`
+}
+```
+
+#### Chains config
+
+The `Chain` abstraction contains all the necessary data to connect to a given chain, query it's state, and send transactions to it. The config will contain an array of these chains (`[]Chain`). Using the `naieve` strategy it will attempt to connect all the chains to all their counterparties. The following data and tools will be needed by each `Chain`:
+
+```go
+type Chain struct {
+    // Key is a reference to either the sdk.AccAddress or key name
+    // which must exist in the keybase associated with this chain.
+    Key   string `yaml:"key"`
+    
+    // The chainID for the chain
+    ChainID       string `yaml:"chain-id"`
+
+    // The sdk.AccAddress prefix for this specific chain
+    AccountPrefix string `yaml:"account-prefix"`
+
+    // HomeDir for this chainID will contain:
+    // - Chain specific keybase
+    // - Lite Client Verifier
+	Verifier      tmlite.Verifier 
+	Keybase       cryptokeys.Keybase 
+
+    // NOTE: is there any weirdness around the codec? Should just the IBC codec suffice?
+    Codec         *codec.Codec
+    
+    // A URL to connect to a node on this chain
+    NodeRPC       string `yaml:"node-rpc"`
+
+    // The list of counterparty chains that this chain should be connected to
+    // NOTE: Relayer will connect and fwd packets for any chains listed here 
+    // which also have an entry in the `[]Chain` 
+    Counterparties []string `yaml:"counterparties"`
+
+    // Gas/Fee settings, are persistent for each transaction on the chain 
+	Gas                uint64       `yaml:"gas"`
+	GasAdjustment      float64      `yaml:"gas-adjustment"`
+	GasPrices          sdk.DecCoins `yaml:"gas-prices"`
+    Fees               sdk.Coins    `yaml:"fees"`
```

Comment:
This is redundant, since fees can be calculated from `gas` and `gasPrices`.

Reply:
These are the options from the TxBuilder. I thought exposing them here would give users the same flexibility on fees as they have for a normal transaction.

Reply:
I see. The thing is that gas amounts will vary a lot depending on what the packet is doing (e.g. sending coins - little gas - or executing a complex delegation state transition - lots of gas). I guess it can still be an option but I doubt it will be useful.

Reply:
However, we should definitely have an option to estimate gas & specify a gas price.